app/institution/submitInstController.js

```diff
@@ -0,0 +1,130 @@
+                .ariaLabel('Confirmar Cadastro')
+                .targetEvent(event)
+                .ok('Sim')
+                .cancel('Não');
+
+            $mdDialog.show(confirm).then(function() {
+                InstitutionService.createInstitution(submitInstCtrl.institution).then(
+                    function success() {
+                        deleteInvite(submitInstCtrl.invite.key).then(
+                            function success() {
+                                goHome();            
+                                showToast('Cadastro de instituição realizado com sucesso');
+                            }, function error(response) {
+                                showToast(response.data.msg);
+                            }
+                        );                    
+                    }, function error(response) {
+                        showToast(response.data.msg);
+                    }
+                );
+            }, function() {
+                showToast('Cancelado');
+            });
+        };
+
+        submitInstCtrl.cancel = function cancel(event) {
+            var confirm = $mdDialog.confirm()
+                .clickOutsideToClose(true)
+                .title('Cancelar Cadastro')
+                .textContent('Cancelar o cadastro dessa instituição?')
+                .ariaLabel('Cancelar Cadastro')
+                .targetEvent(event)
+                .ok('Sim')
+                .cancel('Não');
+
+            $mdDialog.show(confirm).then(function() {
+                deleteInvite(submitInstCtrl.invite.key).then(
+                    function success() {
+                        goHome();            
+                        showToast('Cadastro de instituição cancelado');
+                    }, function error(response) {
+                        showToast(response.data.msg);
+                    }
+                );
+            }, function() {
+                showToast('Cancelado');
+            });
+        };
+
+         var goHome = function goToHome() {
+            $state.go('app.home');
+        };
+
+        function showToast(msg) {
+            $mdToast.show(
+                $mdToast.simple()
+                    .textContent(msg)
+                    .action('FECHAR')
+                    .highlightAction(true)
+                    .hideDelay(5000)
+                    .position('bottom right')
+            );
+        }
+
+        function getLegalNatures() {
+            $http.get('institution/legal_nature.json').then(function success(response) {
+                submitInstCtrl.legalNatures = response.data;
+            });
+        }
+
+        function getOccupationAreas() {
+            $http.get('institution/occupation_area.json').then(function success(response) {
+                submitInstCtrl.occupationAreas = response.data;
+            });
+        }
+
+
+        // TODO: replace the use of this method by the InviteService
```

Comment:
Beleza, muda o replace pra move this method. Por que lá no Service não tem esse ainda. :)